custom_components/samsungtv_tizen/media_player.py

```diff
@@ -375,17 +379,18 @@ def media_title(self):
             if self._cloud_state == STATE_OFF:
                 self._state = STATE_OFF
                 return None
-            elif self._cloud_source in ["digitalTv", "TV"]:
-                if self._cloud_channel_name != "" and self._cloud_channel != "":
-                    if self._show_channel_number:
-                        return self._cloud_channel_name + " (" + self._cloud_channel + ")"
-                    else:
+            else:
+                running_app = self._get_running_app()
```

Comment:
Doing `self._get_running_app()` is a heavy task, because it does 1 request for every app that is set in `app_list`.

I get why the running app is needed here as I just checked and it appears that SmartThings gets stuck on the last input source that was active before starting an app.

But we can't do `self._get_running_app()` in both `.source()` and `.media_title()` because they are both update methods, and `self._get_running_app()` is a lengthy task. So maybe we should do something like:
```
        if self._state == STATE_ON:
            self._running_app = self._get_running_app()
```

at the end of the `update()` method, and then use `self._running_app` everywhere else where we need it?

Reply:
Done.
One thing I noticed though is when `media_title()` reaches
```
return self._source
```
`self._source` may not have been updated yet, so it takes one update longer for the media title to update.

Reply:
Yes, that can happen, we could move all the logic from `.source()` to something like `._get_source()` (as it doesn't use any other update method in the logic anymore), and then do:
```
def source(self):
    return self._get_source()
```
and also use `return self._get_source()` in `media_title()` too?

Reply:
That would mean running `_get_source()` twice per update though.
Theoretically we could do something similar as with `_get_running_app()` before, by calling
```
self._source = self._get_source()
```
at the end of `update()` and then just use `self._source` in both `source()` and `media_title()`

Reply:
We don't need to move `_get_source()` to `update()` because right now, it doesn't rely on any network requests at all, so it's just simplistic logic, which is not a lengthy task at all, so we can abuse of it if we want.